index.js

```diff
@@ -501,25 +525,42 @@ const createFocusTrap = function (elements, userOptions) {
 
       activeFocusTraps.deactivateTrap(trap);
 
-      const onDeactivate =
-        deactivateOptions && deactivateOptions.onDeactivate !== undefined
-          ? deactivateOptions.onDeactivate
-          : config.onDeactivate;
+      const onDeactivate = getOption(deactivateOptions, 'onDeactivate');
+      const onPostDeactivate = getOption(deactivateOptions, 'onPostDeactivate');
+      const checkCanReturnFocus = getOption(
+        deactivateOptions,
+        'checkCanReturnFocus'
+      );
+
       if (onDeactivate) {
         onDeactivate();
       }
 
-      const returnFocus =
-        deactivateOptions && deactivateOptions.returnFocus !== undefined
-          ? deactivateOptions.returnFocus
-          : config.returnFocusOnDeactivate;
+      const returnFocus = getOption(
+        deactivateOptions,
+        'returnFocus',
+        'returnFocusOnDeactivate'
+      );
+
+      const finishDeactivation = () => {
+        if (returnFocus) {
+          delay(() => {
+            tryFocus(getReturnFocusNode(state.nodeFocusedBeforeActivation));
+            if (onPostDeactivate) {
+              onPostDeactivate();
+            }
+          });
+        }
+      };
 
-      if (returnFocus) {
-        delay(function () {
-          tryFocus(getReturnFocusNode(state.nodeFocusedBeforeActivation));
-        });
+      if (checkCanReturnFocus) {
```

Comment:
I think we need to honor the `returnFocus` option here and do `if (returnFocus && checkCanReturnFocus)`